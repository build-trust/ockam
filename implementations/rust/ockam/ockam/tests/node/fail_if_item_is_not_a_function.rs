// This test checks that #[ockam_macros::node] causes a compile time error
// if the item it is defined on is not a function.

#[ockam_macros::node]
struct A {}
