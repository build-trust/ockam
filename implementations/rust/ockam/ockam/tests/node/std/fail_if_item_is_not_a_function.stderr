error: expected `fn`
 --> tests/node/std/fail_if_item_is_not_a_function.rs:2:1
  |
2 | struct A {}
  | ^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/node/std/fail_if_item_is_not_a_function.rs:2:12
  |
2 | struct A {}
  |            ^ consider adding a `main` function to `$DIR/tests/node/std/fail_if_item_is_not_a_function.rs`
